File: id/src/main/java/com/simprints/id/activities/front/FrontContract.java
package com.simprints.id.activities.front;


import android.content.Context;

import com.simprints.id.activities.BasePresenter;
import com.simprints.id.activities.BaseView;

interface FrontContract {

    interface View extends BaseView<Presenter> {

        void setSyncUnavailable();

        void setSyncInProgress();

        void setSyncSuccess();

        void setSyncFailed();
    }

    interface Presenter extends BasePresenter {

        void sync(Context appContext);

        void stopListening();
    }
}

File: id/src/main/java/com/simprints/id/activities/front/FrontContract.kt
package com.simprints.id.activities.front


import com.simprints.id.activities.BasePresenter
import com.simprints.id.activities.BaseView
import com.simprints.id.data.secure.SecureDataManager

interface FrontContract {

    interface View : BaseView<Presenter> {
        fun openRequestAPIActivity()
    }

    interface Presenter : BasePresenter {
        val secureManager: SecureDataManager
        fun doSecurityChecks(): Boolean
    }
}
